src/mgr.sol

```diff
@@ -128,43 +131,43 @@ contract TinlakeManager {
 
     // Tinlake components
     GemLike      public gem;
-    GemLike      public tin;
+    GemLike      public rwaToken;
     RedeemLike   public pool;
 
+    // MIP21 RWAUrn
+    MIP21UrnLike public urn;
+
     uint256 public constant dec = 18;
 
     address public tranche;
 
-    constructor(address vat_,      address dai_,
-                address daiJoin_,  address vow_,
+    constructor(address dai_,      address rwaToken_,
                 address drop_,     address pool_,
                 address operator_, address tranche_,
                 address end_,      bytes32 ilk_
                 ) public {
 
-        vat = VatLike(vat_);
         dai = GemLike(dai_);
         end = EndLike(end_);
-        daiJoin = JoinLike(daiJoin_);
-        vow = vow_;
-
         gem = GemLike(drop_);
+        rwaToken = GemLike(rwaToken_);
+
+
         require(gem.decimals() == dec, "TinlakeMgr/decimals-dont-match");
+        require(rwaToken.decimals() == dec, "TinlakeMgr/decimals-dont-match");
         pool = RedeemLike(pool_);
 
         ilk = ilk_;
         wards[msg.sender] = 1;
         emit Rely(msg.sender);
-
         operator = operator_;
 
         safe = true;
-        live = true;
         glad = true;
+        live = false;
 
-        dai.approve(daiJoin_, uint256(-1));
-        vat.hope(daiJoin_);
-
+        // rwaUrn allowance for rwaToken
+        rwaToken.approve(address(urn), uint(-1));
```

Comment:
this needs to happen in `file` as well?

Reply:
We could make the approval in the `lock` method for the specific amount

Reply:
I think it's fine to move to `file`.

Reply:
Putting it into file is riskier because it requires `urn` to be set which it might not be at the time. Given `lock` is only executed once it doesn't add any inefficiency.